jina/executors/decorators.py

```diff
@@ -255,11 +274,76 @@ def arg_wrapper(*args, **kwargs):
         return _batching
 
 
-def _get_size(data: Union[Iterator[Any], List[Any], np.ndarray], axis: int = 0) -> int:
-    if isinstance(data, np.ndarray):
-        total_size = data.shape[axis]
-    elif hasattr(data, '__len__'):
-        total_size = len(data)
+def batching_multi_input(func: Callable[[Any], np.ndarray] = None, *,
+                         batch_size: Union[int, Callable] = None,
+                         num_batch: Optional[int] = None,
+                         split_over_axis: int = 0,
+                         merge_over_axis: int = 0,
+                         slice_on: int = 1,
+                         num_data: int) -> Any:
+    """Split the input of a function into small batches and call :func:`func` on each batch
+    , collect the merged result and return. This is useful when the input is too big to fit into memory
+
+    :param func: function to decorate
+    :param batch_size: size of each batch
+    :param num_batch: number of batches to take, the rest will be ignored
+    :param split_over_axis: split over which axis into batches
+    :param merge_over_axis: merge over which axis into a single result
+    :param slice_on: the location of the data. When using inside a class,
+            ``slice_on`` should take ``self`` into consideration.
+    :param num_data: the number of data inside the arguments (starting from ``slice_on``)
```

Comment:
> starting from ``slice_on``
This should goes to `.. warning:: `
```suggestion
    :param num_data: the number of data inside the arguments (starting from ``slice_on``)
```

Reply:
Done!